maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -301,11 +307,17 @@ void initialize(CliRequest cliRequest) throws ExitException {
                 System.err.format("-D%s system property is not set.", MULTIMODULE_PROJECT_DIRECTORY);
                 throw new ExitException(1);
             }
-            File basedir = basedirProperty != null ? new File(basedirProperty) : new File("");
-            try {
-                cliRequest.multiModuleProjectDirectory = basedir.getCanonicalFile();
-            } catch (IOException e) {
-                cliRequest.multiModuleProjectDirectory = basedir.getAbsoluteFile();
+            File basedir = new File(basedirProperty);
+            if (new File(basedir, DOT_MVN).isDirectory()) {
+                try {
+                    cliRequest.multiModuleProjectDirectory = basedir.getCanonicalFile();
+                } catch (IOException e) {
+                    cliRequest.multiModuleProjectDirectory = basedir.getAbsoluteFile();
+                }
+            }
+        } else {
+            if (!isMultiModuleProjectDirectoryAcceptable(cliRequest.multiModuleProjectDirectory)) {
+                cliRequest.multiModuleProjectDirectory = null;
```

Comment:
Will be ok that multiModuleProjectDirectory will be null.?
What other impact it can have?

Reply:
This PR corrects this: if not found .mvn folder, it will be null, yes, before it could have any contents. This makes it more consistent than before, when it could contain any path...